modules/internal/String.chpl

```diff
@@ -1115,15 +1115,21 @@ module String {
     proc _getView(r:range(?)) where r.idxType == byteIndex {
       if boundsChecking {
         if r.hasLowBound() && (!r.hasHighBound() || r.size > 0) {
-          if r.low:int <= 0 then
-            halt("range out of bounds of string");
+          if r.low:int < 0 then
+            halt("range ", r, " out of bounds for string");
         }
         if r.hasHighBound() && (!r.hasLowBound() || r.size > 0) {
-          if (r.high:int < 0) || (r.high:int > this.len) then
-            halt("range out of bounds of string");
+          // This seems suspicious... why would a range whose high bound
+          // was -1 be in-bounds but one whose high bound was -2 be out?
+          // (This pre-dated this PR, though the numbers were 0 and -1).
+          // I think that this exists in order to permit the doReplace()
+          // call to work when `find()` returns 0, but this doesn't
+          // seem principled.  See also the similar case in Bytes.chpl.
```

Comment:
I am not sure what is particularly un-principled here. I haven't written this particular logic but I'd be up for changing it.

Reply:
(We discussed offline and I've pushed an update to the comment to try and clarify it a bit.